management/server/http/posture_checks_handler.go

```diff
@@ -230,6 +235,25 @@ func validatePostureChecksUpdate(req api.PostureCheckUpdate) error {
 		}
 	}
 
+	if geoLocationCheck := req.Checks.GeoLocationCheck; geoLocationCheck != nil {
+		if geoLocationCheck.Action == "" {
+			return status.Errorf(status.InvalidArgument, "action for geolocation check shouldn't be empty")
+		}
+		if len(geoLocationCheck.Locations) == 0 {
+			return status.Errorf(status.InvalidArgument, "locations for geolocation check shouldn't be empty")
+		}
+
+		for _, loc := range geoLocationCheck.Locations {
+			if loc.CountryCode == "" {
```

Comment:
Maybe we can add basic validation to check the `2-letter ISO 3166-1 alpha-2 code that represents the country`. Later we can validate against location DB.